app/services/UserService.ts

```diff
@@ -0,0 +1,47 @@
+import {inject, injectable} from 'inversify';
+import {User} from '../persistance/entities/User';
+import {UserRepository} from '../persistance/repositories/UserRepository';
+
+@injectable()
+export class UserService {
+
+  private userRepository: UserRepository;
+
+  constructor(
+      @inject(UserRepository) userRepository: UserRepository
+  ) {
+      this.userRepository = userRepository;
+  }
+
+  public findAll(): Promise<User[]> {
+    return this.userRepository.findAll();
```

Comment:
En cada uno de estos podrías utilizar async / await
`
  public async findOneOrFail(id: string): Promise<User> {
    return await this.userRepository.findOneOrFail(id);
  }
`
Entonces ahi ya te retorna el user o la exception. (Ver si consideraste algun errorHandler o utilizas los try catch en el controller.